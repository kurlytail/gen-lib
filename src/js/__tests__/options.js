import Getopt from 'node-getopt';

import Options from '../options';

jest.mock('node-getopt');

describe('# options', () => {
    describe('## constructor', () => {
        it('### should parse all options', () => {
            Getopt.mockImplementation(() => ({
                bindHelp: jest.fn(),
                parse: jest.fn(() => ({
                    options: { someOption: {} }
                }))
            }));
            let options = new Options({});
            expect(options).toMatchSnapshot();
        });
    });
});
