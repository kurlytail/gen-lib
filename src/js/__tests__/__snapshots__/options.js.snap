// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# options ## constructor ### should parse all options 1`] = `
Options {
  "design": Array [],
  "map": Array [],
  "someOption": Object {},
}
`;
